hub/hubclient/hubclient.go

```diff
@@ -0,0 +1,131 @@
+package hubclient
+
+import (
+	"context"
+
+	"github.com/mailru/easyjson"
+)
+
+// Client interface for interacting with the gateway hub
+type Client interface {
+	// Distributors returns a list of cluster servers on demand
+	Distributors(context.Context) ([]*Server, error)
+	// Servers returns a list of cluster servers on demand
+	Servers(context.Context) ([]*Server, error)
+	// WatchServers returns a channel that will receive serverUpdates updates and a cancel function to end the watch
+	WatchServers() (<-chan []*Server, context.CancelFunc)
+	// WatchDistributors returns a channel to watch for distributor updates and a cancel function to end the watch
+	WatchDistributors() (<-chan []*Server, context.CancelFunc)
+	// WatchConfig returns a channel that will receive config updates and a cancel function to end the watch
+	WatchConfig() (<-chan *Config, context.CancelFunc)
+}
+
+// Server represents a SmartGateway Server
+// easyjson:json
+type Server struct {
+	// Name is the name of the server
+	Name string `json:"name"`
+	// Version is the smart gateway version of the server
+	Version string `json:"version"`
+	// Payload is the payload for the server
+	Payload []byte `json:"payload"`
+	// LastHeartbeat is the timestamp of the last heartbeat
+	LastHeartbeat int `json:"lastHeartbeat"`
+}
+
+// String is the too ToString method for Server
```

Comment:
"too ToString" - I think there's an extra word in the comment

Reply:
whoops yeah that's not right